### PR TITLE
2803983519 create slack section channels

### PR DESCRIPTION
**Pull request by samklq**

### CODE REVIEW

File: student/library/slack.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class Slack:
+    def __init__(self):
+        self.client = WebClient(token=settings.SLACK_USER_OAUTH_TOKEN)
+
+    def create_channel(self, name):
+        try:
+            api_response = self.client.conversations_create(
+                name=name
+            )
+            print(api_response)
```
Comment (rocketkai) [inline student/library/slack.py:15]: Does this `print` need to be left in code?

File: student/library/slack.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        except SlackApiError as error:
+            print(f'Error creating channel: {error}')
```
Comment (rocketkai) [inline student/library/slack.py:17]: Could we also report any errors to Sentry such that we can detect them in our monitoring?

File: staff/views/batch.py [hunk 3]
```diff
@@ -219,3 +216,14 @@ def new_batch_schedules(batch, batch_schedule_formset):
             )
 
     return new_batch_schedules
+
+def set_up_section(batch, section_number, section_capacity):
+    slack_client = Slack()
+    slack_channel_name = f"{batch.number}-{section_number}"
```
Comment (rocketkai) [inline staff/views/batch.py:222]: Could we double-check the preferred naming with @Nuuggs ?

File: learn/settings.py [hunk 4]
```diff
@@ -215,6 +215,11 @@
 
 HUBSPOT_API_KEY = env('HUBSPOT_API_KEY')
 
+# SLACK
+# ------------------------------------------------------------------------------
+
+SLACK_USER_OAUTH_TOKEN = env('SLACK_USER_OAUTH_TOKEN')
```
Comment (rocketkai) [inline learn/settings.py:221]: Should we add a comment to explain that this is Rocket's admin user OAuth token?

**Review by rocketkai — COMMENTED** [review]
Nice 💪

File: student/library/slack.py (hunk 2 above)
Comment (samklq) [reply student/library/slack.py:17]: yep haven't started on sentry story yet, it's in backlog

File: staff/views/batch.py (hunk 3 above)
Comment (samklq) [reply staff/views/batch.py:222]: we decided not to create section channels so early. we will create a `<batch_number>-all` at this point channel instead

File: student/library/slack.py (hunk 1 above)
Comment (samklq) [reply student/library/slack.py:15]: i will replace these when i integrate sentry

**Pull request merged**